unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Catalog/SceneObjectCatalogController.cs

```diff
@@ -0,0 +1,389 @@
+            SetQuickBarShortcut(sceneObject, lastIndexDroped, texture);
+        }
+        Destroy(draggedObject);
+    }
+
+    #endregion
+
+    void SetQuickBarShortcut(SceneObject sceneObject, int index,Texture texture)
+    {     
+        quickBarShortcutsSceneObjects[index] = sceneObject;
+        if (index < shortcutsImgs.Length)
+        {
+            shortcutsImgs[index].texture = texture;
+            shortcutsImgs[index].enabled = true;
+        }
+    }
+
+    #region Favorites
+
+    public void ShowFavorites()
+    {
+        isFavoriteFilterActive = !isFavoriteFilterActive;
+
+        if (isFavoriteFilterActive)
+        {
+            string favoriteName = "Favorites";
+            List<Dictionary<string, List<SceneObject>>> favorites = new List<Dictionary<string, List<SceneObject>>>();
+            foreach (SceneAssetPack assetPack in CatalogController.sceneObjectCatalog.GetValues().ToList())
+            {
+                foreach (SceneObject sceneObject in assetPack.assets)
+                {
+                    foreach (SceneObject favObject in favoritesSceneObjects)
+                    {
+                        if (favObject != null)
+                        {
+                            if (sceneObject.id == favObject.id && sceneObject.asset_pack_id == favObject.asset_pack_id)
+                            {
+                                bool foundCategory = false;
+                                foreach (Dictionary<string, List<SceneObject>> groupedSceneObjects in favorites)
+                                {
+                                    if (groupedSceneObjects.ContainsKey(favoriteName))
+                                    {
+                                        foundCategory = true;
+                                        if (!groupedSceneObjects[favoriteName].Contains(sceneObject))
+                                            groupedSceneObjects[favoriteName].Add(sceneObject);
+                                    }
+                                }
+                                if (!foundCategory)
+                                {
+                                    Dictionary<string, List<SceneObject>> groupedSceneObjects = new Dictionary<string, List<SceneObject>>();
+                                    groupedSceneObjects.Add(favoriteName, new List<SceneObject>() { sceneObject });
+                                    favorites.Add(groupedSceneObjects);
+                                }
+                                break;
+
+                            }
+                        }
+                    }
+                }
+            }
+            catalogTitleTxt.text = favoriteName;
+            ShowCatalogContent();
+            catalogGroupListView.SetContent(favorites);
+        }
+        else
+        {
+            ShowAssetsPacks();
+        }
+       
+    }
+    public void AsignFavorite(SceneObject sceneObject, CatalogItemAdapter adapter)
```

Comment:
This is a bit confusing, the methods seems to toggle the favorite state. We should find a better naming

Reply:
`ToggleFavoriteState` sounds better?

Reply:
Perfect!